plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java

```diff
@@ -4898,19 +4925,8 @@ public boolean editEntityIdentifier() {
         EditVirtualEntityDialog dialog = new EditVirtualEntityDialog(
             ResultSetViewer.this, model.getSingleSource(), model.getVirtualEntity(true));
         dialog.setInitPage(EditVirtualEntityDialog.InitPage.UNIQUE_KEY);
-        if (dialog.open() == IDialogConstants.OK_ID) {
-            DBDRowIdentifier virtualID = getVirtualEntityIdentifier();
-            if (virtualID != null) {
-                try {
-                    virtualID.reloadAttributes(new VoidProgressMonitor(), getModel().getAttributes());
-                } catch (DBException e) {
-                    log.error(e);
-                }
-            }
-            persistConfig();
-            return true;
-        }
-        return false;
+        // Removed redundant logic: identifier reload and config persistence are handled inside dialog.okPressed()
```

Comment:
If you want to left a comment for reviewers, you can do it in the GitHub. Such comments become useless as soon as PR merged.